rpcs3/Emu/Cell/Modules/cellSysutil.cpp

```diff
@@ -72,8 +72,18 @@ extern void sysutil_send_system_cmd(u64 status, u64 param)
 
 struct syscache
 {
-	atomic_t<u32> state = 0;
-	std::string cache_path;
+	atomic_t<bool> mounted = false;
+	std::string cache_id;
+	shared_mutex mtx;
+
+	~syscache()
+	{
+		// Check if mounted and cache_id is different than already written empty value
+		if (!cache_id.empty())
+		{
+			fs::write_file(fs::get_cache_dir() + "/cache/cacheId", fs::rewrite, cache_id);
```

Comment:
Maybe theres a better alternative to the way its storing last cache id from the previous boot?